quinn-proto/src/cid_generator.rs

```diff
@@ -31,6 +31,33 @@ pub trait ConnectionIdGenerator: Send + Sync {
     fn cid_lifetime(&self) -> Option<Duration>;
 }
 
+/// Validates connection id in short header packet
+pub trait ConnectionIdValidator {
+    /// Validate and regconize a connection id from given buffer, return the length of connection id
+    /// or [`None`] if it is invalid
+    fn validate(&self, buffer: &[u8]) -> Option<usize>;
```

Comment:
If we're going to do this, I think we should deduplicate this with `ConnectionIdGenerator::validate()` somehow?

Reply:
See https://github.com/quinn-rs/quinn/pull/1865#issuecomment-2117104025